CMakeLists.txt

```diff
@@ -37,3 +47,23 @@ add_subdirectory(src/go)
 add_subdirectory(src/c)
 
 install(DIRECTORY "pixmaps" DESTINATION "${PURPLE_DATA_DIR}" FILES_MATCHING PATTERN "*.png")
+
+# package settings
+set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "Whatsapp protocol plug-in for libpurple with pidgin")
+set(CPACK_PACKAGE_VENDOR "Bryant Eadon")
+set(CPACK_PACKAGE_DESCRIPTION "whatsmeow is a go-based whatsapp protocol plug-in for libpurple based on the whatsapp business apis.")
+set(CPACK_PACKAGE_CONTACT "bryant@balancebitsconsulting.com")
+set(CPACK_PACKAGE_VERSION_MAJOR "${VERSION_MAJOR}")
+set(CPACK_PACKAGE_VERSION_MINOR "${VERSION_MINOR}")
+set(CPACK_PACKAGE_VERSION_PATCH "${VERSION_PATCH}")
+set(CPACK_DEBIAN_PACKAGE_SHLIBDEPS ON)
+set(CPACK_DEBIAN_PACKAGE_ARCHITECTURE "${ARCHITECTURE}")
+set(CPACK_SOURCE_PACKAGE_FILE_NAME "${CMAKE_PROJECT_NAME}_${VERSION}_${CPACK_DEBIAN_PACKAGE_ARCHITECTURE}")
+set(CPACK_PACKAGE_FILE_NAME "${CMAKE_PROJECT_NAME}_${VERSION}_${CPACK_DEBIAN_PACKAGE_ARCHITECTURE}")
+set(CPACK_DEBIAN_PACKAGE_DEPENDS "libpurple0 (>= 2.10.0), libglib2.0-0 (>= 2.24), libjson-glib-1.0-0 (>= 0.8.0)")

```

Comment:
`libjson-glib-1.0-0` is not a dependency of this project. 

However, `gdk-pixbuf-2.0` and `opusfile` are optional dependencies which may or may not be present depending on what was available at time of cmake configuraion. This would need to be reflected in the list of dependencies.

Also, I am not sure about the versions. There are a couple of version switches in the code:

* GLIB_CHECK_VERSION in bridge.c and handle_attachment.c
* PURPLE_VERSION_CHECK in init.c handle_attachment.c
* maybe others somewhere else

These, too, affect (minimum) dependency versions during build time.

Reply:
Ok, then that means we are getting more complex in the building and packaging.  I would need to detect the build instructions and package different based on these dependencies. This could be done.

However, to make this a easier, I propose making `gdk-pixbuf-2.0` and `opusfile` requirements of the project.

Otherwise, if this kind of option is required, this balloons the number of packages from 1 to 4.  I struggle to see the value of publishing 4 different versions per platform ( without, with pixbuf, with opusfile, with pixbuf + opusfile ).  Then, arm , another 4 packages, arch, another 4  .. windows, another 4 ... then multiple this by the versions , and test every one on all these platforms.

If we think about this critically are there *real* active use cases where there is a need to leave these (optional) dependencies out ?

Reply:
I can easily remove the dependencies that are not required "libjson-glib-1.0-0 ".

Reply:
For the versions, I now use an approach that relies on the version information supplied by pkg-config.

The availability of optional libraries is checked, the dependencies are expanded conditionally.

Reply:
Ok, let's see how this goes.

I am concerned about the testing load this persists.  I would like to simplify packaging / building to a single set of dependencies long term.